tests/homework2/test_task4.py

```diff
@@ -0,0 +1,49 @@
+from unittest import mock
+
+import pytest
+
+from homework2.task4 import cache
+
+
+def getArgStr(args, kwargs):
+    return str(args) + ", " + str(kwargs)
+
+
+args = [(5, 6), (2, "someStr", [2, 3, 4])]
+kwargs = [{"k": 7}, {"qwert": [2, 3], "a": 9}]
+results = [3, 7]
+
+values = dict(zip((map(getArgStr, args, kwargs)), results))
+
+
+def side_effect(args, kwargs):
+    return values[getArgStr(args, kwargs)]
+
+
+@pytest.mark.parametrize(
+    "arguments, kwargs",
+    [
+        (args[0], kwargs[0]),
+        (args[1], kwargs[1]),
+        (args[0], kwargs[0]),
+        (args[1], kwargs[1]),
+        (args[0], kwargs[0]),
+    ],
+)
+def test_cache_several_arguments(arguments, kwargs):
```

Comment:
This test doesn't show that mock function is called once per every pair of args and kwargs because you create a new mock function for each pair of test parameters.
Could you call a function several times in one test run with different args and kwargs?